// Copyright 2021 Google LLC. All Rights Reserved.
// 
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
// 
//     http://www.apache.org/licenses/LICENSE-2.0
// 
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
syntax = "proto3";
// All generated protos should be opaque, with "xxx_hidden_" prepended to their field names.


package dcl;

import "proto/connector/sdk.proto";
import "proto/empty.proto";

enum NetworkconnectivityAlphaSpokeStateEnum {
  NetworkconnectivityAlphaSpokeStateEnumNO_VALUE_DO_NOT_USE = 0;
  NetworkconnectivityAlphaSpokeStateEnumSTATE_UNSPECIFIED = 1;
  NetworkconnectivityAlphaSpokeStateEnumCREATING = 2;
  NetworkconnectivityAlphaSpokeStateEnumACTIVE = 3;
  NetworkconnectivityAlphaSpokeStateEnumDELETING = 4;
}

message NetworkconnectivityAlphaSpoke {
  string name = 1;
  string create_time = 2;
  string update_time = 3;
  map<string, string> labels = 4;
  string description = 5;
  string hub = 6;
  repeated string linked_vpn_tunnels = 7;
  repeated string linked_interconnect_attachments = 8;
  repeated NetworkconnectivityAlphaSpokeLinkedRouterApplianceInstances linked_router_appliance_instances = 9;
  string unique_id = 10;
  NetworkconnectivityAlphaSpokeStateEnum state = 11;
  string project = 12;
  string location = 13;
}

message NetworkconnectivityAlphaSpokeLinkedRouterApplianceInstances {
  string virtual_machine = 1;
  string ip_address = 2;
}

message ApplyNetworkconnectivityAlphaSpokeRequest {
  NetworkconnectivityAlphaSpoke resource = 1;
  repeated LifecycleDirective lifecycle_directives = 2;
  string service_account_file = 3;
}

message DeleteNetworkconnectivityAlphaSpokeRequest {
  string service_account_file = 1;
  NetworkconnectivityAlphaSpoke resource = 2;
}

message ListNetworkconnectivityAlphaSpokeRequest {
  string service_account_file = 1;
  string Project = 2;
  string Location = 3;
}

message ListNetworkconnectivityAlphaSpokeResponse {
  repeated NetworkconnectivityAlphaSpoke items = 1;
}

service NetworkconnectivityAlphaSpokeService {
  rpc ApplyNetworkconnectivityAlphaSpoke(ApplyNetworkconnectivityAlphaSpokeRequest) returns (NetworkconnectivityAlphaSpoke);
  rpc DeleteNetworkconnectivityAlphaSpoke(DeleteNetworkconnectivityAlphaSpokeRequest) returns (google.protobuf.Empty);
  rpc ListNetworkconnectivityAlphaSpoke(ListNetworkconnectivityAlphaSpokeRequest) returns (ListNetworkconnectivityAlphaSpokeResponse);
}
